java-frontend/src/main/java/org/sonar/java/regex/ast/RegexSource.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.regex.ast;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+
+/**
+ * The source code of a regular expression, made up out of 1 or more string literals
+ */
+public class RegexSource {
+
+  private final List<LiteralTree> stringLiterals;
+  private final String sourceText;
+  private final TreeMap<Integer, Integer> indices;
+
+  public RegexSource(List<LiteralTree> stringLiterals) {
+    this.stringLiterals = stringLiterals;
```

Comment:
I wonder about this `List`. since we are passing the reference, it might be possible to modify it after. We probably better provide a `Collections.unmodifiableList(...)`.

Reply:
@m-g-sonar Are you worried that `RegexSource` might modify the list? Or that someone else (with access to the original list) might modify the list after the `RegexSource` is created? Because `unmodifiableList` won't prevent the latter - we'd need to make an actual copy to protect against that. And as for the former, I don't think it's necessary to use `unmodifiableList` to prevent modifications from within the same class.

Reply:
Actually you're right, we should be making it unmodifiable because the literals are accessible through a public getter. I forgot about that. Thanks.

Reply:
Though on third thought, it might make more sense to just remove the getter because I don't think we actually need it.